ocaml/xapi/xapi_vm_lifecycle.ml

```diff
@@ -941,6 +942,8 @@ let force_state_reset_keep_current_operations ~__context ~self ~value:state =
       (Db.PCI.get_all ~__context)
   ) ;
   update_allowed_operations ~__context ~self ;
+  if old_state <> state && (old_state = `Running || state = `Running) then
```

Comment:
I'm not sure if how this would be impacted by the transient `paused` state?

Reply:
If there is a transient state change, such as from paused to running or from running to paused, I believe we should attempt to generate or dismiss an alert accordingly.
Based on my testing, I have not encountered this issue. during the VM lifecycle there is just one shot to try to generate alert.